apps/next15-demo/src/app/Story.tsx

```diff
@@ -0,0 +1,23 @@
+"use client";
+import React from "react";
+import { StoryblokComponent } from "./StoryblokComponent";
+import type * as StoryblokComponents from "@/lib/types";
+import { useStory } from "@/lib/useStory";
+
+type Component =
+  | NonNullable<StoryblokComponents.DefaultPage["body"]>[number]
+  | StoryblokComponents.DefaultPage
+  | StoryblokComponents.HeadlineSegment
+  | StoryblokComponents.Button
+  | StoryblokComponents.NavItem
+  | StoryblokComponents.ArticlePage;
+
+export function Story({ slug }: { slug: string }) {
+  const { story, error, isLoading } = useStory<Component>(slug);
```

Comment:
This is a user hook that wraps the their data fetching library (SWR in this case) using the js client. Since the SWR cache is already pre-populated in the Server Component, this will simply return the data needed to complete SSR.

On the client, the SWR cache is re-hydrated and bridge updates will mutate the cache and trigger live updates.